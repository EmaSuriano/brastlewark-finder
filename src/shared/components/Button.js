import styled from 'styled-components';
import PropTypes from 'prop-types';

const DISABLED_FUNC = () => false;

const Button = styled.button.attrs({
  disabled: props => props.disabled,
  onClick: props => (props.disabled ? DISABLED_FUNC : props.onClick),
})`
  font-size: 1em;
  padding: 0.5em;
  margin: 0.5em;
  background: ${props =>
    props.disabled ? props.theme.disabled : props.theme.accent};
  border-radius: 3px;
  border: none;
  color: ${props =>
    props.disabled ? props.theme.secondary : props.theme.light};
  cursor: ${props => (props.disabled ? 'not-allowed' : 'pointer')};
`;

Button.propTypes = {
  disabled: PropTypes.bool,
  onClick: PropTypes.func,
};

Button.defaultProps = {
  disabled: false,
  onClick: DISABLED_FUNC,
};

export default Button;
