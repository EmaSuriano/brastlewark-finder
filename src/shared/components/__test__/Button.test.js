import React from 'react';
import { shallow } from 'enzyme';
import Button from '../Button';

describe('<Button />', () => {
  describe('render', () => {
    it('defaultProps', () => {
      expect(shallow(<Button />)).toMatchSnapshot();
    });

    it('with disabled', () => {
      expect(shallow(<Button disabled />)).toMatchSnapshot();
    });

    it('with onClick and enable, should call onClick on clicking', () => {
      const onClick = jest.fn();
      const button = shallow(<Button onClick={onClick} />);
      button.simulate('click');
      expect(onClick).toHaveBeenCalled();
    });

    xit('with onClick and disable, should not call onClick on clicking', () => {
      const onClick2 = jest.fn();
      const button = shallow(<Button onClick={onClick2} disabled />);
      button.simulate('click');

      expect(onClick2).not.toHaveBeenCalled();
    });
  });
});
