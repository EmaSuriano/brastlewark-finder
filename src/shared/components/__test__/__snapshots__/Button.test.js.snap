// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> render defaultProps 1`] = `
.c0 {
  font-size: 1em;
  padding: 0.5em;
  margin: 0.5em;
  border-radius: 3px;
  border: none;
  cursor: pointer;
}

<button
  className="c0"
  disabled={false}
  onClick={[Function]}
/>
`;

exports[`<Button /> render with disabled 1`] = `
.c0 {
  font-size: 1em;
  padding: 0.5em;
  margin: 0.5em;
  border-radius: 3px;
  border: none;
  cursor: not-allowed;
}

<button
  className="c0"
  disabled={true}
  onClick={[Function]}
/>
`;
